### PR TITLE
Feature/store data

### PR DESCRIPTION
**Pull request by ReAliens**
in this pull request, we implemented the following:- 
- [x] add methods to convert data to JSON format
- [x] add methods to store data in written files with JSON 
- [x] trigger the saving action on app Exit

### CODE REVIEW

File: handlers.rb [hunk 1]
```diff
@@ -80,38 +81,86 @@ def create_book
...
-  print 'ID of person: '
-  id = gets.chomp.to_i
+  def parse_people
+    file = 'people.json'
+    return [] unless File.exist? file
+
+    JSON.parse(File.read(file)).map do |people|
+      if people['json_class'] == 'Student'
+        student = Student.new(age: people['age'], name: people['name'], parent_permission: people['permission'],
+                              classroom: @classroom.label)
```
Comment (dasileker) [inline handlers.rb:140]: - [ ] Nice Job on preserving the data, but we have a slight issue on the re-opening the app it render an error about this particular line The code is parsing a JSON file and creating an array of Student objects.
The code is iterating through the array, adding each object to the @people variable.
The code creates a student object and adds it to the @people array. but the why classroom.label is not in your stored def.

Capture d’écran 2022-03-08 171047

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @ReAliens,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- PR is good👌.
- Linters are passing 👌.
- Readme file is nice 👌.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: handlers.rb (hunk 1 above)
Comment (ReAliens) [reply handlers.rb:140]: thanks for your awesome review

**Review by o-rashad — APPROVED** [review]
Asslam Alaikum Team, @ReAliens @Chimwemwe-127001 ,😊

Hi👋🏽, 
Everything is working fine now 😉👍
Thanks for implementing the changes required by the previous reviewer 👌  
Your project is complete!  
There is nothing else to say other than... it's time to merge it 😎 

# Congratulations! 🎉 

Happy coding!👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ReAliens** [comment]
thanks for your great effort :smile:

**Pull request merged**